src/components/dashboard/WhoContent.native.js

```diff
@@ -0,0 +1,194 @@
+import React, { useCallback, useEffect } from 'react'
+import { FlatList, PermissionsAndroid } from 'react-native'
+import Contacts from 'react-native-contacts'
+import { get, orderBy, uniq } from 'lodash'
+import { isAndroid } from '../../lib/utils/platform'
+import { Section } from '../common'
+import Separator from '../common/layout/Separator'
+import InputText from '../common/form/InputText'
+import { withStyles } from '../../lib/styles'
+import normalize from '../../lib/utils/normalizeText'
+import { getDesignRelativeHeight } from '../../lib/utils/sizes'
+import userStorage from '../../lib/gundb/UserStorage'
+import FeedContactItem from './FeedContactItem'
+
+const WhoContent = ({ styles, setContact, error, text, value, next, state, showNext, setValue }) => {
+  const [contacts, setContacts] = React.useState([])
+  const [initialList, setInitalList] = React.useState(contacts)
+  const [recentFeedItems, setRecentFeedItems] = React.useState([])
+  const [recentlyUsedList, setRecentlyUsedList] = React.useState([])
+
+  const getAllContacts = () => {
+    Contacts.getAll((err, contacts) => {
+      if (err === 'denied') {
+        console.warn('permissions denied')
+      } else {
+        const sortContacts = orderBy(contacts, ['givenName'])
+        setContacts(sortContacts)
+        setInitalList(sortContacts)
+      }
+    })
+  }
+
+  const getUserFeed = async () => {
+    const userFeed = await userStorage.getFeedPage(5, true)
+    const recent = userFeed.filter(({ type }) => type === 'send' || type === 'receive')
+    setRecentFeedItems(recent)
+  }
+
+  const showPermissionsAndroid = () => {
+    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+      title: 'Contacts',
+      message: 'We need access to view your contacts, so you can easily send G$ to them.',
+      buttonPositive: 'Approve',
+    }).then(() => {
+      getAllContacts()
+    })
+  }
+
+  useEffect(() => {
+    getUserFeed()
+  }, [])
+
+  useEffect(() => {
+    if (Contacts) {
+      if (isAndroid) {
+        showPermissionsAndroid()
+      } else {
+        getAllContacts()
+      }
+    }
+  }, [])
+
+  const handleSearch = useCallback(query => {
+    const queryIsNumber = parseInt(query)
+    if (state) {
+      setValue(query)
+    }
+    if (query && !query.includes('+') && !query.includes('*')) {
+      if (typeof queryIsNumber === 'number' && !isNaN(queryIsNumber)) {
+        return setContacts(
+          initialList.filter(({ phoneNumbers }) => get(phoneNumbers, '[0].number', '').indexOf(query) >= 0)
+        )
+      }
+      if (typeof query === 'string') {
+        return setContacts(
+          initialList.filter(({ givenName, familyName }) => {
+            const fullName = `${givenName} ${familyName}`.toLocaleLowerCase()
+            return fullName.indexOf(query.toLocaleLowerCase()) >= 0
+          })
+        )
+      }
+    } else {
+      getAllContacts()
```

Comment:
why you call it here ? you already called it  on component mounted and stored onto initialList state var
so you have full contacts list separately from the filtered contacts

Reply:
Also, as initialList isn't used in jsx and doesn't affects component changes directly, you could store it inside ref